# Tinder Project - Speed Dating 
### What makes people say yes to a second date?

In [101]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import plotly.express as px
import chardet 
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [102]:
file = 'Speed+Dating+Data.csv'

#detect file encoding

with open(file, 'rb') as f:
    data = f.read()

# Detect the encoding
result = chardet.detect(data)
print(result['encoding'])

# Build the Dataframe
data = pd.read_csv(file, encoding=result['encoding'])
df = pd.DataFrame(data)

df.sample(10)

Windows-1252


,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
2228,162,5.0,1,10,2,7,16,15,14.0,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3962,270,19.0,0,37,2,11,21,2,2.0,11,...,6.0,4.0,7.0,6.0,5.0,6.0,5.0,7.0,6.0,5.0
5832,383,5.0,0,9,2,15,19,8,8.0,12,...,7.0,8.0,5.0,6.0,4.0,7.0,8.0,6.0,7.0,5.0
4529,299,6.0,0,11,2,12,14,4,4.0,13,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5205,351,11.0,0,21,2,14,18,11,11.0,2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1198,86,11.0,0,21,2,4,18,7,NaN,12,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1513,104,11.0,1,22,2,4,18,10,NaN,15,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
223,22,2.0,0,3,2,2,16,7,NaN,14,...,7.0,8.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN
7792,526,18.0,0,35,2,21,22,14,14.0,11,...,4.0,3.0,3.0,7.0,7.0,2.0,4.0,4.0,7.0,7.0
2059,152,11.0,0,21,2,7,16,12,12.0,6,...,7.0,8.0,9.0,9.0,8.0,NaN,NaN,NaN,NaN,NaN


In [103]:
df.describe(include='all')

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
count,8378.000000,8377.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,6532.000000,8378.000000,...,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000
unique,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,283.675937,8.960248,0.500597,17.327166,1.828837,11.350919,16.872046,9.042731,9.295775,8.927668,...,7.240312,8.093357,8.388777,7.658782,7.391545,6.810020,7.615079,7.932540,7.155258,7.048611
std,158.583367,5.491329,0.500029,10.940735,0.376673,5.995903,4.358458,5.514939,5.650199,5.477009,...,1.576596,1.610309,1.459094,1.744670,1.961417,1.507341,1.504551,1.340868,1.672787,1.717988
min,1.000000,1.000000,0.000000,1.000000,1.000000,1.000000,5.000000,1.000000,1.000000,1.000000,...,2.000000,2.000000,3.000000,2.000000,1.000000,2.000000,2.000000,4.000000,1.000000,1.000000
25%,154.000000,4.000000,0.000000,8.000000,2.000000,7.000000,14.000000,4.000000,4.000000,4.000000,...,7.000000,7.000000,8.000000,7.000000,6.000000,6.000000,7.000000,7.000000,6.000000,6.000000
50%,281.000000,8.000000,1.000000,16.000000,2.000000,11.000000,18.000000,8.000000,9.000000,8.000000,...,7.000000,8.000000,8.000000,8.000000,8.000000,7.000000,8.000000,8.000000,7.000000,7.000000
75%,407.000000,13.000000,1.000000,26.000000,2.000000,15.000000,20.000000,13.000000,14.000000,13.000000,...,8.000000,9.000000,9.000000,9.000000,9.000000,8.000000,9.000000,9.000000,8.000000,8.000000


In [104]:
df.columns

Index(['iid', 'id', 'gender', 'idg', 'condtn', 'wave', 'round', 'position',
       'positin1', 'order',
       ...
       'attr3_3', 'sinc3_3', 'intel3_3', 'fun3_3', 'amb3_3', 'attr5_3',
       'sinc5_3', 'intel5_3', 'fun5_3', 'amb5_3'],
      dtype='object', length=195)

In [105]:
df.isnull().any().any()

True

# Identify unique member from each wave

In [133]:
unique = df[['iid', 'gender', 'age', 'race', 'goal', 'date', 'go_out', 'attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1',
             'attr3_1']].drop_duplicates('iid')

unique['gender_label'] = unique['gender'].map({0: 'female', 1: 'male'})

race_map = {1: 'Black/African American', 2: 'European/Caucasian', 3: 'Latino/Hispanic',
            4: 'Asian/Pacific Islander', 5: 'Native American', 6: 'Other'}
unique['race_label'] = unique['race'].map(race_map)

goal_map = {1: 'Fun night out', 2: 'Meet new people', 3: 'Get a date',
            4: 'Serious relationship', 5: 'Say I did it', 6: 'Other'}
unique['goal_label'] = unique['goal'].map(goal_map)

unique

,iid,gender,age,race,goal,date,go_out,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr3_1,gender_label,race_label,goal_label
0,1,0,21.0,4.0,2.0,7.0,1.0,15.0,20.0,20.0,15.0,15.0,15.0,6.0,female,Asian/Pacific Islander,Meet new people
10,2,0,24.0,2.0,1.0,5.0,1.0,45.0,5.0,25.0,20.0,0.0,5.0,7.0,female,European/Caucasian,Fun night out
20,3,0,25.0,2.0,6.0,3.0,1.0,35.0,10.0,35.0,10.0,10.0,0.0,8.0,female,European/Caucasian,Other
30,4,0,23.0,2.0,1.0,5.0,1.0,20.0,20.0,20.0,20.0,10.0,10.0,7.0,female,European/Caucasian,Fun night out
40,5,0,21.0,2.0,2.0,4.0,1.0,20.0,5.0,25.0,25.0,10.0,15.0,6.0,female,European/Caucasian,Meet new people
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8268,548,1,30.0,2.0,1.0,6.0,1.0,40.0,10.0,20.0,10.0,0.0,20.0,10.0,male,European/Caucasian,Fun night out
8290,549,1,28.0,2.0,1.0,7.0,1.0,20.0,20.0,20.0,20.0,0.0,20.0,7.0,male,European/Caucasian,Fun night out
8312,550,1,30.0,2.0,2.0,4.0,1.0,30.0,3.0,30.0,30.0,3.0,4.0,7.0,male,European/Caucasian,Meet new people
8334,551,1,27.0,1.0,1.0,6.0,3.0,40.0,20.0,20.0,20.0,0.0,0.0,7.0,male,Black/African American,Fun night out


In [134]:
print(f'Total participants: {len(unique)}')
print(f'Total speed dates recorded: {len(df)}')
print(f'Overall match rate: {df["match"].mean()*100:.1f}%')

Total participants: 551
Total speed dates recorded: 8378
Overall match rate: 16.5%


# 1. Sample Overview

### 1.1. Gender distribution

In [135]:
counts = unique['gender_label'].value_counts().reset_index()
counts.columns = ['Gender', 'Count']

fig = px.pie(
    counts, values='Count', names='Gender',
    title='Gender Distribution of Participants',
    color='Gender',
    hole=0.5
)
 
fig.update_traces(
    texttemplate='%{label}<br>%{value} (%{percent})',
    textposition='outside'
)
 
fig.update_layout(
    showlegend=False,
    title_font_size=18,
    margin=dict(t=60, b=40, l=40, r=40)
)
 
fig.show()



**Caption:** The dataset is nearly perfectly balanced with 274 women (49.7%) and 277 men (50.3%). This balance is by design — each speed dating event was organized with equal numbers of men and women. For our analysis of second date decisions, this means we can compare male and female behavior without worrying about sampling bias.

### 1.2 Age Distribution

In [136]:
f_mean, f_std = unique[unique['gender']==0]['age'].mean(), unique[unique['gender']==0]['age'].std()
m_mean, m_std = unique[unique['gender']==1]['age'].mean(), unique[unique['gender']==1]['age'].std()

In [137]:
fig = px.histogram(
    unique, x='age',facet_col='gender_label',
    nbins=20,
    color='gender_label',
    title='Age Distribution by Gender — Mean and ±1 Standard Deviation',
    labels={'age': 'Age', 'gender_label': 'Gender'}
)

for col, mean, std in [(1, f_mean, f_std),(2, m_mean, m_std)]:
    fig.add_vline(x=mean, line_dash='solid', line_color='Black', line_width=2,
                  annotation_text=f'Mean = {mean:.1f}',
                  annotation_position='top right',
                  annotation_font_color=color, col=col)
    fig.add_vline(x=mean - std, line_dash='dash', line_color='Black', line_width=1.5,
                  annotation_text=f'-1 SD ({mean-std:.1f})',
                  annotation_position='bottom left',
                  annotation_font_color=color, col=col)
    fig.add_vline(x=mean + std, line_dash='dash', line_color='Black', line_width=1.5,
                  annotation_text=f'+1 SD ({mean+std:.1f})',
                  annotation_position='bottom right',
                  annotation_font_color=color, col=col)    

fig.update_layout(showlegend=False)

fig.show()


**Caption:** Most participants are between 22 and 30 years old, reflecting the graduate student population at Columbia University. 

### 1.3 Race Distribution

In [138]:
counts_race = unique['race_label'].value_counts().reset_index()
counts_race.columns = ['Race', 'Count']

fig = px.bar(counts_race,
             x='Count', y='Race',
             title='Racial Composition of Participants',
             color='Count',
             color_continuous_scale='Blues',
             text='Count',             
)

fig.update_traces(textposition='outside')
fig.update_layout(
    coloraxis_showscale=False,
    yaxis=dict(categoryorder='total ascending'),
    xaxis_title='Number of participants',
    yaxis_title='',
    title_font_size=16
)

fig.show()

European/Caucasian participants are the largest group (304 out of 545 with valid data), followed by Asian/Pacific Islander (136). The sample reflects the demographics of Columbia University's graduate programs in the early 2000s. 

### 1.4 Participant Goals

In [139]:
counts_goal = unique['goal_label'].value_counts().reset_index()
counts_goal.columns = ['Goal', 'Count']
counts_goal['Percentage'] = (counts_goal['Count'] / counts_goal['Count'].sum() * 100).round(1)

fig = px.bar(
    counts_goal, x='Goal', y='Count',
    title='Why did participants sign up?',
    color='Count',
    color_continuous_scale='Pinkyl',
    text=counts_goal['Percentage'].apply(lambda x: f'{x}%')
)
fig.update_traces(textposition='inside')
fig.update_layout(
    coloraxis_showscale=False,
    xaxis_title='', yaxis_title='Number of participants',
    title_font_size=16
)
fig.show()

The majority of participants came for a fun night out (41.4%) or to meet new people (34.3%). Only 4% explicitly sought a serious relationship. This is important context for interpreting second date decisions: most "yes" responses are driven by casual interest rather than long-term intentions

## 2. Match Rates

### 2.1 Overall Decision and Match Rate

In [140]:
yes_rate_f = df[df['gender'] == 0]['dec'].mean() * 100
yes_rate_m = df[df['gender'] == 1]['dec'].mean() * 100
match_rate = df['match'].mean() * 100

fig = px.bar(df,x=['Women say yes', 'Men say yes', 'Mutual match'],
             y=[yes_rate_f, yes_rate_m, match_rate],
             color=['#D4537E', '#378ADD', '#1D9E75'],
             text=[f'{yes_rate_f:.1f}%', f'{yes_rate_m:.1f}%', f'{match_rate:.1f}%'],
             )

fig.update_traces(textposition='inside')

fig.update_layout(
    title='Decision and Match Rates',
    yaxis_title='Rate (%)', yaxis_range=[0, 60],
    xaxis_title='',
    title_font_size=16,
    showlegend=False
)
fig.show()

Women said yes to 36.5% of their partners, while men said yes to 47.4%. In fact, men are significantly more likely to agree to a second date. Yet despite these individual rates, only 16.5% of encounters result in a mutual match. This gap highlights the core challenge: agreeing to meet again requires both people to feel the same way, which is relatively rare. The lower female selectivity threshold acts as the binding constraint on mutual matches.

### 2.2 Match Rate by Race Pairing

In [141]:
same_race_match = df.groupby('samerace')['match'].mean().reset_index()

same_race_match['same_race_label'] = same_race_match['samerace'].map({0: 'Different race', 1: 'Same race'})
same_race_match['match_pct'] = (same_race_match['match'] * 100).round(1)

fig = px.bar(same_race_match,
             x = 'same_race_label',
             y = 'match_pct',
             text=same_race_match['match_pct'].apply(lambda x: f'{x}%'),
             color='same_race_label',
             color_discrete_map={'Same race': '#1D9E75', 'Different race': '#888780'}

)

fig.show()

Same-race pairs have a match rate of 17.1% vs 16.1% for different-race pairs : a small but present gap. The effect is subtle enough that racial similarity alone is not a strong predictor of second date agreement. Other factors, particularly perceived attractiveness and fun, play a far larger role in the decision.

## 3. What Actually Drives the "Yes"?

### 3.1 Average Attribute Ratings by Match Outcome

In [142]:
attrs = ['attr', 'fun', 'shar', 'intel', 'sinc', 'amb']
attr_labels = ['Attractive', 'Fun', 'Shared interests', 'Intelligent', 'Sincere', 'Ambitious']

attr_match = df.groupby('match')[attrs].mean().round(2)

fig = go.Figure()
fig.add_trace(go.Bar(
    name='No match', x=attr_labels,
    y=attr_match.loc[0][attrs].values,
    marker_color='#B4B2A9'
))
fig.add_trace(go.Bar(
    name='Match', x=attr_labels,
    y=attr_match.loc[1][attrs].values,
    marker_color='#1D9E75'
))
fig.update_layout(
    barmode='group',
    title='Average Partner Ratings: Match vs No Match',
    yaxis_title='Average rating (1–10)',
    yaxis_range=[0, 10],
    xaxis_title='',
    title_font_size=16,
    legend_title='Outcome'
)
fig.show()


Across all six attributes, partners who result in a mutual match receive substantially higher ratings. The largest gaps appear for **attractiveness** (+1.36 points) and **fun** (+1.44 points), while **sincerity** and **intelligence** show smaller gaps (+0.76 and +0.70). This suggests that second date decisions are driven primarily by how attractive and fun someone seems in 4 minutes rather than deeper character assessments like sincerity or ambition.

### 3.2 Correlation of Ratings with Personal Decision

In [143]:
corr = df[attrs + ['dec']].corr()['dec'].drop('dec').reset_index()
corr.columns = ['Attribute', 'Correlation']
corr['Attribute'] = attr_labels

corr = corr.sort_values('Correlation', ascending=True)

corr['color'] = corr['Correlation'].apply(lambda x: '#1D9E75' if x >= 0.4 else '#5DCAA5' if x >= 0.2 else '#9FE1CB')

fig = px.bar(
    corr, x='Correlation', y='Attribute', orientation='h',
    title='Correlation Between Partner Ratings and Saying "Yes"',
    text=corr['Correlation'].apply(lambda x: f'{x:.3f}')
)
fig.update_traces(
    marker_color=corr['color'].values,
    textposition='outside'
)
fig.update_layout(
    xaxis_title='Pearson correlation with decision (yes=1)',
    yaxis_title='',
    xaxis_range=[0, 0.6],
    title_font_size=16
)
fig.show()

Attractiveness (r = 0.487) is the single strongest predictor of agreeing to a second date, followed by fun (r = 0.414) and shared interests (r = 0.401). Intelligence (r = 0.217) and sincerity (r = 0.210) correlate far less with the decision, and ambition (r = 0.184) is the weakest predictor. This ranking reveals that snap judgments about desirability and chemistry in a 4-minute interaction are dominated by surface-level appeal, not the deeper qualities people often claim to prioritize.

### 3.3 Stated Preferences vs Actual Predictors (by Gender)

In [159]:
pref_cols = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1']
pref_labels = ['Attractive', 'Sincere', 'Intelligent', 'Fun', 'Ambitious', 'Shared interests']

pref_gender = unique.groupby('gender_label')[pref_cols].mean().round(2)
actual_corr = df[attrs + ['dec']].corr()['dec'].drop('dec').values * 100

fig = make_subplots(
    rows=1, cols=2,
    subplot_titles=('What people say they want (100pt allocation)', 'What actually predicts saying yes (corr × 100)')
)

fig.add_trace(go.Bar(
    name='Women', x=pref_labels, y=pref_gender.loc['female'].values,
    marker_color='#D4537E'
), row=1, col=1)
fig.add_trace(go.Bar(
    name='Men', x=pref_labels, y=pref_gender.loc['male'].values,
    marker_color='#378ADD'
), row=1, col=1)

fig.add_trace(go.Bar(
    name='Actual predictor', x=pref_labels,
    y=actual_corr,
    marker_color='#1D9E75', showlegend=True
), row=1, col=2)

fig.update_layout(
    title='Stated Preferences vs What Actually Drives Decisions',
    barmode='group',
    title_font_size=16,    
    legend_title=''
)
    

fig.show()

There is a clear gap between what people *say* they want and what actually drives their decisions. Men allocate 27.3 points to attractiveness while women allocate 18.0 points. Both groups rank it highly in stated preferences. However, when we look at what actually predicts the "yes" decision (right panel), attractiveness dominates even more strongly (correlation of 0.487), while sincerity and intelligence — which both genders say they value — barely move the needle. People systematically understate how much appearance drives their split-second dating decisions.

## 4. Context and Situational Effects

### 4.1 Decision Fatigue: Match Rate by Meeting Order

In [169]:
order_match = df.groupby('order').agg(
    match_rate=('match', 'mean'),
    dec_rate=('dec', 'mean'),
    n=('match', 'count')
).reset_index()

order_match = order_match[order_match['n'] > 50]

fig = go.Figure()
fig.add_trace(go.Scatter(
    x=order_match['order'], y=(order_match['dec_rate'] * 100).round(1),
    mode='lines+markers', name='Personal yes rate',
    line=dict(color='#378ADD', width=2),
    marker=dict(size=6)
))
fig.add_trace(go.Scatter(
    x=order_match['order'], y=(order_match['match_rate'] * 100).round(1),
    mode='lines+markers', name='Mutual match rate',
    line=dict(color='#1D9E75', width=2),
    marker=dict(size=6)
))
fig.update_layout(
    title='Decision and Match Rate by Order of Meeting',
    xaxis_title='Order of date during the evening',
    yaxis_title='Rate (%)',
    title_font_size=16,
    legend_title=''
)
fig.show()

Both the personal yes rate and the mutual match rate decline as the evening progresses. Participants met in the first 5 encounters have a match rate of 18%, falling to less than 12% for those met 21st or later. This decision fatigue effect suggests that context, not just the person's qualities, shapes second date decisions. Meeting someone early in the night, when energy and openness are higher, is a genuine advantage.

### 4.2 Self-Rating vs Received Rating (Ego Calibration)

In [179]:
self_ratings = unique[['iid', 'attr3_1', 'gender_label']].dropna()
received = df.groupby('pid')['attr_o'].mean().reset_index()

received.columns = ['iid', 'avg_received']

calibration = self_ratings.merge(received, on='iid')

calibration['gap'] = calibration['attr3_1'] - calibration['avg_received']

fig = go.Figure()
for gender, color in [('female', '#D4537E'), ('male', '#378ADD')]:
    sub = calibration[calibration['gender_label'] == gender]
    fig.add_trace(go.Histogram(
        x=sub['gap'], name=gender,
        marker_color=color, opacity=0.7,
        nbinsx=20
    ))
fig.add_vline(x=0, line_dash='dash', line_color='gray', annotation_text='Perfectly calibrated')
fig.update_layout(
    title='Self-Rated Attractiveness minus Received Attractiveness Rating',
    xaxis_title='Gap (positive = overestimate)',
    yaxis_title='Number of participants',
    barmode='overlay',
    title_font_size=16,
    legend_title='Gender'
)
fig.show()

Most participants overestimate their attractiveness, with the distribution skewed right of zero. This systematic overconfidence has implications for second date decisions: people who overestimate their own desirability may be more selective, reducing their match rate by saying no to partners who would have been a realistic match.

## 5. Liking, Probability Estimates, and Match

### 5.1 Overall Liking Score Distribution by Match Outcome

In [187]:
like_data = df[['like', 'match']].dropna()
like_data['outcome'] = like_data['match'].map({0: 'No match', 1: 'Match'})

fig = px.violin(
    like_data, x='outcome', y='like', color='outcome',
    box=True, points='outliers',
    title='"How much did you like this person?" by Match Outcome',
    color_discrete_map={'No match': '#B4B2A9', 'Match': '#1D9E75'},
    labels={'like': 'Liking score (1–10)', 'outcome': ''}
)
fig.update_layout(
    showlegend=False,
    yaxis_title='Liking score (1–10)',
    title_font_size=16
)
fig.show()

The overall liking score is strongly linked to match outcomes. For matched pairs, liking scores cluster between 7 and 10, while non-matched pairs have a much wider and lower distribution. The "liking" question captures a gut feeling about the person and it correlates with match outcomes (r = 0.306) more than most individual attribute ratings. This suggests that when people decide to agree to a second date, they are reacting to an overall impression rather than consciously weighing individual traits.

### 5.2 Perceived Probability of Partner's Interest

In [190]:
prob_data = df[['prob', 'match']].dropna()
prob_data['outcome'] = prob_data['match'].map({0: 'No match', 1: 'Match'})

fig = px.box(
    prob_data, x='outcome', y='prob', color='outcome',
    title='Estimated Probability that Partner Will Say Yes',
    color_discrete_map={'No match': '#B4B2A9', 'Match': '#1D9E75'},
    labels={'prob': 'Estimated probability (%)', 'outcome': ''}
)
fig.update_layout(
    showlegend=False,
    title_font_size=16
)
fig.show()


Participants were asked to estimate the probability that their partner would say yes to them. For matched pairs, this estimate is higher (median ~60%) than for non-matched pairs (~50%). This shows that people have some intuition about mutual attraction : they can partially sense when the feeling is reciprocated. The correlation between perceived probability and match outcome (r = 0.256) indicates that social perceptiveness during a 4-minute interaction is a real, measurable signal for second date success.

## 6. Summary

In [195]:
summary = pd.DataFrame({
    'Factor': [
        'Partner attractiveness rating',
        'Partner fun rating',
        'Shared interests rating',
        'Overall liking score',
        'Perceived partner interest',
        'Partner intelligence rating',
        'Partner sincerity rating',
        'Same race',
        'Age gap',
        'Meeting order (early vs late)'
    ],
    'Correlation with match': [0.487, 0.414, 0.401, 0.306, 0.256, 0.217, 0.210, 0.010, -0.05, -0.09],
    'Direction': ['Strong +', 'Strong +', 'Strong +', 'Moderate +', 'Moderate +',
                  'Weak +', 'Weak +', 'Negligible', 'Slight −', 'Slight −']
})

fig = px.bar(
    summary.sort_values('Correlation with match'),
    x='Correlation with match', y='Factor', orientation='h',
    title='Summary: What Predicts Agreeing to a Second Date?',
    color='Correlation with match',
    color_continuous_scale='Teal',
    text='Correlation with match'
)
fig.update_traces(texttemplate='%{text:.2f}', textposition='outside')
fig.update_layout(
    coloraxis_showscale=False,
    xaxis_title='Correlation with match outcome',
    yaxis_title='',
    title_font_size=16
)
fig.show()


The summary chart consolidates our findings. The top predictors of a mutual second date agreement are **attractiveness, fun, and shared interests** all perceived in the moment during the 4-minute interaction. Deeper qualities like intelligence and sincerity, as well as demographic similarity (race, age), play a much smaller role. Situational factors like meeting order also matter, though modestly. The core finding is that second date decisions in speed dating are driven predominantly by immediate chemistry and physical appeal, even when participants claim to prioritize character over looks.